### DISCUSSION TITLE
[Feature] Custom Metadata

### DISCUSSION THREAD

**Discussion by TheDarthFather** [paperless-ngx/paperless-ngx#453] [Feature Requests]
Currently we have 2 metadata "objects" (Corespondent, Document type) and multiple tags, wich is great.
An option will be to let each admin to create additional metadata "objects" and add multiple values for each.
This is allow 100% flexibility for any use-case. 

PS: The tags may still exist as is a easy way to filter and to visualize (I love the colored "stamp" effect on documents), and the exiting 2 can let to exist as default examples.   

Later EDIT: Realised later there is a discussion on this topic.

**Comment by KarstenDE** [comment]
That would be very great!

I think different type of such custom fields would be helpful:
- Objects with one selection (like today the document type)
- Objects with multiple selections (like today the tags)
- Datetime Information
- Freetext with single line
- Freetext with multiple lines (for different things incl a log diary etc)

Possibly it make sense to add such custom fields to a seperate tab at the individual document

**Reply by esclear** [reply]
The freetext fields could also offer optional verification, for example using regular expressions.

**Comment by jtommi** [comment]
I was thinking of using https://github.com/invoice-x/invoice2data to fill custom metadata like Invoice amount, invoice date, invoice number, etc through the API

**Reply by mtettke123** [reply]
conbined with custom metadata fields a real great idea. Would improve my invoice-workflow so much!

**Comment by SaschaWeisenberger** [comment]
+1 for single line freetext (searchable of course). Default invoice fields would be awesome, but in the first step configuring additional fields would be enough for me.

**Comment by patsch9** [comment]
++1 as example for an contract id. The simplest way is to add custom namebale freetext fields.

**Comment by Mannshoch** [comment]
I need a field for a time-range so I could set invoices to specific time range and check if I have all time range scanned.

Is there a way to have tags and fields included into PDF?

**Comment by sam-wiltshire** [comment]
The ability to import documents + metadata in JSON/csv/text format which could be assigned to custom metadata fields.

If using with some form of document automation would make it a powerful DMS
